templates/commands/render_test.go

```diff
@@ -505,10 +592,29 @@ steps:
 			if diff := cmp.Diff(gotDestContents, tc.wantDestContents, cmpFileMode); diff != "" {
 				t.Errorf("dest directory contents were not as expected (-got,+want): %s", diff)
 			}
+
+			gotBackupContents := loadDirWithoutMode(t, backupDir)
+			gotBackupContents = stripFirstPathElem(gotBackupContents)
+			if diff := cmp.Diff(gotBackupContents, tc.wantBackupContents, cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("backups directory contents were not as expected (-got,+want): %s", diff)
+			}
 		})
 	}
 }
 
+// Since os.MkdirTemp adds an extra random token, we strip it back out to get
+// determistic results.
+func stripFirstPathElem(m map[string]string) map[string]string {
```

Comment:
Hmm yea I agree this is a bit weird, but wouldn't we have the same issue if we used the randint?

Reply:
It's a lot simpler with the randint. You just generate the path you *would* use at startup time. You do this at the top-level, in realRun(). Then when it's time to create a backup file, you just MkdirAll the folder to create-if-not-exists. For the lower-level functions like copyRecursive and TestCopyRecursive, the backup directory is just a string to them. The tests just write to t.TempDir(), and the real prod code takes a directory name from the higher level that happens to have a randint in it.

Reply:
Let me try to explain that better: MkdirAll is idempotent, using a randint that was generated at startup. MkdirTemp isn't idempotent, so we have to ensure it only runs once.

Reply:
Hmm okay - I'm fine if we:

1. Document this super duper well
2. Return a very specific error if the directory already exists